src/github_utils.py

```diff
@@ -0,0 +1,132 @@
+import json
+import os
+import requests
+from github import Github
+
+def get_pull_request_diff():
+    """Fetch the pull request diff content from GitHub"""
+    g = Github(os.getenv('GIT_TOKEN'))
+
+    # Parse GitHub event context
+    event_path = os.getenv('GITHUB_EVENT_PATH')
+    with open(event_path, 'r') as f:
+        event_data = json.load(f)
+
+    repo = g.get_repo(event_data['repository']['full_name'])
+    pr_number = event_data['pull_request']['number']
+    pull_request = repo.get_pull(pr_number)
+
+    # Get diff content
+    diff_url = pull_request.diff_url
+    response = requests.get(diff_url)
+    return response.text
+
+
+def get_valid_lines(diff_content):
+    """Parse diff to find valid line numbers with proper initialization"""
+    valid_lines = {}
+    current_file = None
+    line_number = None
+    
+    for line in diff_content.split('\n'):
+        # Reset tracking for new files
+        if line.startswith('diff --git'):
+            current_file = line.split(' b/')[-1].strip()
+            valid_lines[current_file] = set()
+            line_number = None  # Reset for new file
+            continue
+            
```

Comment:
Initializing `line_number` inside the `if line.startswith('diff --git'):` is a good start, but consider initializing it outside the for loop as well to prevent errors when the diff doesn't start with a 'diff --git' line.

```suggestion
line_number = None # initialize before loop
```